initialization/metadata.go

```diff
@@ -17,14 +19,27 @@ func SaveMetadata(dir string, v *shared.PostMetadata) error {
 		return fmt.Errorf("dir creation failure: %w", err)
 	}
 
-	data, err := json.Marshal(v)
+	filename := filepath.Join(dir, MetadataFileName)
+
+	tmp, err := os.Create(fmt.Sprintf("%s.tmp", filename))
 	if err != nil {
-		return fmt.Errorf("serialization failure: %w", err)
+		return fmt.Errorf("create temporary file %s: %w", tmp.Name(), err)
 	}
+	defer tmp.Close()
 
-	err = os.WriteFile(filepath.Join(dir, MetadataFileName), data, shared.OwnerReadWrite)
-	if err != nil {
-		return fmt.Errorf("write to disk failure: %w", err)
+	enc := json.NewEncoder(tmp)
+	enc.SetIndent("", "\t")
+
+	if err := enc.Encode(v); err != nil {
+		return fmt.Errorf("failed to encode metadata: %w", err)
+	}
+
+	if err := tmp.Close(); err != nil {
+		return fmt.Errorf("failed to close tmp file %s: %w", tmp.Name(), err)
+	}
+
+	if err := atomic.ReplaceFile(tmp.Name(), filename); err != nil {
+		return fmt.Errorf("atomic replace file %s with %s: %w", filename, tmp.Name(), err)
 	}
```

Comment:
How about using an in-memory `bytes.Buffer` instead of a temporary file?
```go
	buf := new(bytes.Buffer)
	enc := json.NewEncoder(buf)
	enc.SetIndent("", "\t")

	if err := enc.Encode(v); err != nil {
		return fmt.Errorf("failed to encode metadata: %w", err)
	}
	if err := atomic.WriteFile(filepath.Join(dir, MetadataFileName), buf); err != nil { ... }
```

Reply:
Wanted to be efficient, but we can do in memory as well. The file is small and not updated that often.

Reply:
btw. the only difference between `atomic.WriteFile` and `atomic.ReplaceFile` is that `atomic.WriteFile` first writes to a temporary file created by `os.CreateTemp` and then calls `atomic.ReplaceFile(tmp, dest)`.

Reply:
Yeah but the code is simpler which is the most important in this case :)